src/psyclone/tests/psyir/symbols/datatype_test.py

```diff
@@ -415,6 +417,8 @@ def test_arraytype_unsupportedtype():
     # Since no partial datatype is provided, these return None
     assert utype.partial_datatype is None
     assert utype.intrinsic is None
+    # Test the allocatable flag
+    assert not utype.is_allocatable
```

Comment:
As suggested in the implementation, I think this should return None in this case (c.f. the `intrinsic` property).

Reply:
Fixed.